src/main/java/hudson/plugins/dimensionsscm/CallbackInstance.java

```diff
@@ -0,0 +1,31 @@
+package hudson.plugins.dimensionsscm;
+
+
+import com.serena.dmclient.api.DimensionsConnection;
+
+public class CallbackInstance {
+
+    private static DimensionsAPICallback dimensionsAPICallback;
+
+    private CallbackInstance() {}
+
+    public static DimensionsAPICallback getInstance(DimensionsConnection connection) {
+
+        if (dimensionsAPICallback != null)
+            return dimensionsAPICallback;
+
+        if (isNewServerVersion(connection))
+            dimensionsAPICallback = new DimensionsAPICallback14();
+        else
+            dimensionsAPICallback = new DimensionsAPICallback12();
+
+        return dimensionsAPICallback;
+    }
+
+    private static boolean isNewServerVersion(DimensionsConnection connection) {
```

Comment:
I'd call this something like `supportsChangeSets` instead, because the plugin supports multiple versions of Dimensions, and there may be other features we test for using the server version in future